### DISCUSSION TITLE
[Bug] When i uses filter to build condition in oal, the filter expressions do not support linking by and, or

### DISCUSSION THREAD

**Discussion by tjzheng1002** [apache/skywalking#10536] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

When i uses filter to build condition in oal, the filter expressions does not support linking by and, or.The documentation says it supports it, but I've tried it.

**The test oal：**
`test_service_error_count = from(ServiceProblem.*).filter(type == ProblemType.ERROR and tag["org"] == "team").count();`
![image](https://user-images.githubusercontent.com/6240715/224925808-447ce4f8-f662-4d76-929c-f6ef28516eca.png)
**The documentation：**
![image](https://user-images.githubusercontent.com/6240715/224926320-6236bb70-a58f-40bb-ad6b-2aef120974cc.png)

But it can support like this:
`test_service_error_count = from(ServiceProblem.*).filter(type == ProblemType.ERROR) .filter(tag["org"] == "team").count();`

### What you expected to happen

Support and,or in oal.

### How to reproduce

When i uses filter to build condition in oal, the filter expressions does not support linking by and, or.
**The test oal：**
`test_service_error_count = from(ServiceProblem.*).filter(type == ProblemType.ERROR and tag["org"] == "team").count();`
![image](https://user-images.githubusercontent.com/6240715/224925808-447ce4f8-f662-4d76-929c-f6ef28516eca.png)

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
I think that is generally speaking. It relies on function implementations(OPs) to determine the final results.
